src/coreclr/jit/gentree.cpp

```diff
@@ -31452,6 +31478,48 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
             }
 #endif
 
+            case NI_Vector128_op_Equality:
+#if defined(TARGET_ARM64)
+            case NI_Vector64_op_Equality:
+#elif defined(TARGET_XARCH)
+            case NI_Vector256_op_Equality:
+            case NI_Vector512_op_Equality:
+#endif // !TARGET_ARM64 && !TARGET_XARCH
+            {
+                if (varTypeIsFloating(simdBaseType))
```

Comment:
how is this path different from `case NI_Vector128_op_Equality:` above? if it's only for floats, then why it's not an assert?

Reply:
Ah, it's when only one of the operand is constant?

Reply:
Right, it’s for when one is all nan, which is an optimization we can do for float/double